src/array.jl

```diff
@@ -972,3 +949,33 @@ Base.repeat(a::CatArrOrSub{T, N},
 Base.repeat(a::CatArrOrSub{T, N};
             inner = nothing, outer = nothing) where {T, N} =
     CategoricalArray{T, N}(repeat(refs(a), inner=inner, outer=outer), copy(pool(a)))
+
+# JSON3 writing/reading
+StructTypes.StructType(::Type{<:CategoricalVector}) = StructTypes.ArrayType()
+
+function StructTypes.construct(::Type{<:CategoricalArray}, A::Vector)
+    if eltype(A) == Any
+        categorical(identity.(ifelse.(A .=== nothing, missing, A)))
```

Comment:
@quinnj Do this approach sound correct to you?

Reply:
That looks fine to me